modules/Movecraft/src/main/java/net/countercraft/movecraft/sign/StatusSign.java

```diff
@@ -84,50 +85,109 @@ public final void onSignTranslate(SignTranslateEvent event) {
         }
         int signLine=1;
         int signColumn=0;
-        /*  TODO: Implement new system for flyblocks on status signs
-        for(EnumSet<Material> alFlyBlockID : craft.getType().getFlyBlocks().keySet()) {
-            Material flyBlockID= alFlyBlockID.get(0);
-            double minimum=craft.getType().getFlyBlocks().get(alFlyBlockID).get(0);
-            if(foundBlocks.get(flyBlockID) != 0 && minimum>0) { // if it has a minimum, it should be considered for sinking consideration
-                int amount=foundBlocks.get((flyBlockID));
-                double percentPresent= (amount*100D/totalBlocks);
-                String signText="";
-                if(percentPresent>minimum*1.04) {
-                    signText+= ChatColor.GREEN;
-                } else if(percentPresent>minimum*1.02) {
-                    signText+=ChatColor.YELLOW;
-                } else {
-                    signText+=ChatColor.RED;
-                }
-                if(flyBlockID == Material.REDSTONE_BLOCK) {
-                    signText+="R";
-                } else if(flyBlockID == Material.IRON_BLOCK) {
-                    signText+="I";
-                } else {
-                    signText+= flyBlockID.toString().charAt(0);
-                }
+        //The following code is cannibalized from AsyncManager but I'm not sure of a better way to do it
```

Comment:
I'm not sure if it is in the scope of this PR, but it might be a wise decision to make a cache of the results of such a calculation.

Reply:
If we're doing that, we might as well go the full distance and store the data when it is calculated by AsyncManager, to stop status signs on crafts being such a massive performance hit. Do you think that'd be a good idea?

Reply:
I'm not sure if you want it to be part of this PR or not, but my assumption was that you'd include that in the caching.